src/core/core.scale.js

```diff
@@ -909,27 +930,65 @@ var Scale = Element.extend({
 	/**
 	 * @private
 	 */
-	_isVisible: function() {
+	_getMatchingMetas: function() {
 		var me = this;
 		var chart = me.chart;
-		var display = me.options.display;
+		var axisID = me.isHorizontal() ? 'xAxisID' : 'yAxisID';
+		var result = [];
 		var i, ilen, meta;
 
+		for (i = 0, ilen = chart.data.datasets.length; i < ilen; ++i) {
+			if (!chart.isDatasetVisible(i)) {
+				continue;
+			}
+			meta = chart.getDatasetMeta(i);
+			if (meta[axisID] === me.id) {
+				meta._datasetIndex = i;
+				result.push(meta);
+			}
+		}
+		return result;
+	},
+
+	/**
+	 * @private
+	 */
+	_isVisible: function() {
+		var display = this.options.display;
+
 		if (display !== 'auto') {
 			return !!display;
 		}
 
-		// When 'auto', the scale is visible if at least one associated dataset is visible.
-		for (i = 0, ilen = chart.data.datasets.length; i < ilen; ++i) {
-			if (chart.isDatasetVisible(i)) {
-				meta = chart.getDatasetMeta(i);
-				if (meta.xAxisID === me.id || meta.yAxisID === me.id) {
-					return true;
+		return this._getMatchingMetas().length > 0;
+	},
+
+	_getStackKey: function(meta) {
+		return [
+			meta.type,
+			// we have a separate stack for stack=undefined datasets when the opts.stacked is undefined
+			((this.options.stacked === undefined && meta.stack === undefined) ? meta._datasetIndex : ''),
+			meta.stack
+		].join('.');
+	},
+
+	/**
+	 * @private
+	 */
+	_hasStacks: function(metas) {
+		var result = this.options.stacked;
+		var i, ilen;
+
+		metas = metas || this._getMatchingMetas();
```

Comment:
I wonder if it'd fix a code climate warning if you inverted the `if` check:

```		metas = metas || this._getMatchingMetas();
		if (result !== undefined) {
			return result;
		}
		for (i = 0, ilen = metas.length; i < ilen; ++i) {
			if (metas[i].stack !== undefined) {
				return true;
			}					
		}					

		return result;
```

Reply:
Also dropped the fallback to `this._getMatchingMetas()`, because its not used that way. (at 10 warnings now)